### PR TITLE
Refactor/mstore

### PR DESCRIPTION
**Pull request by Eikix**
Refactor mstore to use store_n 

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-
-
-

### CODE REVIEW

File: src/kakarot/instructions/memory_operations.cairo [hunk 1]
```diff
@@ -100,7 +101,14 @@ namespace MemoryOperations {
         let offset = popped[1];
         let value = popped[0];
 
-        let memory: model.Memory* = Memory.store(self=ctx.memory, element=value, offset=offset.low);
+        
+        let (bytes_array_len, bytes_array) = Helpers.uint256_to_bytes_array(value);
+
+        with_attr error_message("Uint256 represented as an array of bytes should be of length 32") {
+            assert bytes_array_len = 32;
+        }
```
Comment (ClementWalter) [inline src/kakarot/instructions/memory_operations.cairo:109]: this is more a test of the helpers than something required here right?

File: tests/cairo_files/test_memory.cairo [hunk 2]
```diff
@@ -108,18 +121,6 @@ func test__load__should_load_an_element_from_the_memory_with_offset{
     return ();
 }
 
-@external
-func test__load__should_fail__when_out_of_memory{
```
Comment (ClementWalter) [inline tests/cairo_files/test_memory.cairo:112]: what happened to this test? though I don't really understand it, there's no out of memory in the EVM right?

Comment (Eikix) [reply tests/cairo_files/test_memory.cairo:112]: yeah that's my understanding

File: src/kakarot/instructions/memory_operations.cairo (hunk 1 above)
Comment (Eikix) [reply src/kakarot/instructions/memory_operations.cairo:109]: yeah fair

**Review by ClementWalter — APPROVED** [review]

**Pull request merged**